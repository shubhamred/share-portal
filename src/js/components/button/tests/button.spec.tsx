import React from 'react';
import { shallow, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Button from '../button';

configure({ adapter: new Adapter() });

describe('components/button', () => {
  it('Renders default options', () => {
    const component = shallow(
      <Button onClick={jest.fn()} />
    );
    const tree = shallowToJson(component);
    expect(tree).toMatchSnapshot();
  });
  it('Renders disabled options', () => {
    const component = shallow(
      <Button disabled={true} onClick={jest.fn()} />
    );
    const tree = shallowToJson(component);
    expect(tree).toMatchSnapshot();
  });
  describe('Check for click handle', () => {
    const props = {
      disabled: false,
      label: 'dummy',
      onClick: jest.fn()
    };
    // eslint-disable-next-line react/jsx-props-no-spreading
    const ButtonComponentWrapper = shallow(<Button {...props} />);

    it('check prop type children should not be empty', () => {
      expect(ButtonComponentWrapper.props().children).toBeTruthy();
    });

    it('should handle click', () => {
      ButtonComponentWrapper.simulate('click');
    });
  });
});
