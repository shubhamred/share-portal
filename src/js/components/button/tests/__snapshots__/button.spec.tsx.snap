// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/button Renders default options 1`] = `
<button
  className="btn"
  disabled={false}
  onClick={[Function]}
  style={Object {}}
  title=""
  type="button"
/>
`;

exports[`components/button Renders disabled options 1`] = `
<button
  className="disabled"
  disabled={true}
  onClick={[Function]}
  style={Object {}}
  title=""
  type="button"
/>
`;
